### ISSUE TITLE
[Finder] Checking existence of file in DateRangeFilterIterator doesn't work properly with FTP

### ISSUE THREAD

**Issue by davidkmenta** [symfony/symfony#16429]
Hi!

`SplFileInfo::getRealPath` function returns `false` if file does not exist and apparently it also returns `false` for FTP path (eg: "ftp://username:password@hostname/filename.xml") even if file exists.
It could be probably fixed by using `SplFileInfo::getPathname` function instead.

**Comment by dolbertz** [comment]
I confirmed on PHP 5.6.18, that `SplFileInfo::getRealPath()` does return `false` for an FTP path:
```
$splFileInfo = new SplFileInfo('ftp://username:password@hostname/filename.xml');
var_dump($splFileInfo->getRealPath());
var_dump($splFileInfo->getPathname());
```

This returns `false` for the first var_dump call and returns `'ftp://username:password@hostname/filename.xml'` for the second one.

`SplFileInfo::getRealPath()` is used in the DateRangeFilterIterator on this line:
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Finder/Iterator/DateRangeFilterIterator.php#L47

So, I added this two lines to the end of my test, because DateRangeFilterIterator calls `file_exists()` on `SplFileInfo::getRealPath()`:
```
var_dump(file_exists($splFileInfo->getRealPath()));
var_dump(file_exists($splFileInfo->getPathname()));
```

This returns `false` and `true`.
 
But I'm not sure if a simple switch from `getRealPath()` to `getPathname()` in DateRangeFilterIterator would work in all intended cases. Maybe an additional check, if this is in fact a local file access, or not, would be neccessary, when the functionality of this code relies on absolute paths.
